src/main/java/net/snowflake/ingest/utils/ParameterProvider.java

```diff
@@ -17,13 +17,19 @@ public class ParameterProvider {
   public static final String ENABLE_SNOWPIPE_STREAMING_METRICS_MAP_KEY =
       "ENABLE_SNOWPIPE_STREAMING_JMX_METRICS".toLowerCase();
 
+  public static final String BLOB_FORMAT_VERSION = "BLOB_FORMAT_VERSION".toLowerCase();
+
   // Default values
   public static final long BUFFER_FLUSH_INTERVAL_IN_MILLIS_DEFAULT = 1000;
   public static final long BUFFER_FLUSH_CHECK_INTERVAL_IN_MILLIS_DEFAULT = 100;
   public static final long INSERT_THROTTLE_INTERVAL_IN_MILLIS_DEFAULT = 500;
   public static final long INSERT_THROTTLE_THRESHOLD_IN_PERCENTAGE_DEFAULT = 10;
   public static final boolean SNOWPIPE_STREAMING_METRICS_DEFAULT = false;
 
+  public static final int BLOB_FORMAT_VERSION_DEFAULT = 1;
+  public static final int BLOB_FORMAT_START_SUPPORTED_VERSION = 1;
+  public static final int BLOB_FORMAT_END_SUPPORTED_VERSION = 2;
+
```

Comment:
Yes, the idea was to support two formats for some time, in case of any noticed regressions for the newer format. Server side PR supports both versions anyway to handle the older format even now in PrPr.